codebase/entity_connexion.py

```diff
@@ -90,65 +150,70 @@ def get_transactions(self):
+                                                    edge_attr='amount', create_using=nx.DiGraph())
         if len(transaction_graph) > 40:  # if too many nodes, no point in plotting
             warnings.warn('Too many nodes, not producing graph')
             warnings.warn('Too many nodes, not producing graph')
             return
 
         # determine receiver/sender/trader status
         attrs = {}
-        trans_entities = set(df['transferringEntity_id'])
-        acqui_entities = set(df['acquiringEntity_id'])
+        trans_entities = set(transaction_table['transferringEntity_id'])
+        acqui_entities = set(transaction_table['acquiringEntity_id'])
 
         # colors
         color_legend = {'this': 'green', 'trader': 'violet', 'sender': 'blue', 'receiver': 'red'}
         color_handles = []
         for c in color_legend:
             color_handles.append(mpatches.Patch(color=color_legend[c], label=c))
 
+        entity_id_to_name = self.map_entity_id_to_name(transaction_table)
+
         # loop over nodes
         for node in transaction_graph:
             attrs[node] = {}
             # todo: change color based on account type not on trader_type
+            attrs[node]['name'] = entity_id_to_name[node]
+            attrs[node]['id'] = node
             if (node in trans_entities) and (node in acqui_entities):
                 attrs[node]['trader_type'] = 'trader'
                 attrs[node]['color'] = color_legend[attrs[node]['trader_type']]
-                attrs[node]['type'] = df[df['transferringEntity_id'] == node].iloc[0]['transferringEntity_type']
-                attrs[node]['name'] = df[df['transferringEntity_id'] == node].iloc[0]['transferringEntity_name']
-                attrs[node]['id'] = df[df['transferringEntity_id'] == node].iloc[0]['transferringEntity_id']
+                # TODO: commented out the line below because there is no "Type" column in the AccountHolder table.
+                #  -> is there such a thing as a "type" attribute for AccountHolders of Companies ?
+                # attrs[node]['type'] = df[df['transferringEntity_id'] == node].iloc[0]['transferringEntity_type']
             elif node in trans_entities:
                 attrs[node]['trader_type'] = 'sender'
                 attrs[node]['color'] = color_legend[attrs[node]['trader_type']]
-                attrs[node]['type'] = df[df['transferringEntity_id'] == node].iloc[0]['transferringEntity_type']
-                attrs[node]['name'] = df[df['transferringEntity_id'] == node].iloc[0]['transferringEntity_name']
-                attrs[node]['id'] = df[df['transferringEntity_id'] == node].iloc[0]['transferringEntity_id']
+                # TODO: commented out the line below because there is no "Type" column in the AccountHolder table.
+                #  -> is there such a thing as a "type" attribute for AccountHolders of Companies ?
```

Comment:
No just for accounts but that we would like to use for the colors, no? I would keep it.

Reply:
Ok, defined (empty) fields `acquiringEntity_type` and `transferringEntity_type` for `AccountHolders`.